File: 0901-1000/0946-Validate Stack Sequences/0946-Validate Stack Sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        stack<int> St;
        int j = 0;
        for (int i : pushed) {
            St.push(i);
            while (!St.empty() && St.top() == popped[j]) {
                St.pop();
                ++j;
            }
        }
        return St.empty();
    }
};

File: 0901-1000/0946-Validate Stack Sequences/0946-Validate Stack Sequences.py
class Solution:
    def validateStackSequences(self, pushed: 'List[int]', popped: 'List[int]') -> 'bool':
        St = []
        j = 0
        for i in pushed:
            St.append(i)
            while St and St[-1] == popped[j]:
                St.pop()
                j += 1
        return len(St) == 0
